src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,8 +1,54 @@
 package core.basesyntax;
 
-public class WorkWithFile {
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
 
+public class WorkWithFile {
     public void getStatistic(String fromFileName, String toFileName) {
+        int[] data = parseData(getData(fromFileName));
+        String report = createReport(data);
+        writeReport(toFileName, report);
+    }
+
+    private List<String> getData(String fileName) {
+        List<String> allLines;
+        try {
+            allLines = Files.readAllLines(Path.of(fileName));
+        } catch (IOException e) {
+            throw new RuntimeException("Can't read file " + fileName, e);
+        }
+        return allLines;
+    }
+
+    private void writeReport(String fileName, String report) {
+        try {
+            Files.createFile(Path.of(fileName));
+            Files.write(Path.of(fileName), report.getBytes());
+        } catch (IOException e) {
+            throw new RuntimeException("Can't write to file " + fileName, e);
+        }
+    }
+
+    private String createReport(int[] data) {
+        int result = data[0] - data[1];
+        return "supply," + data[0] + System.lineSeparator()
+                + "buy," + data[1] + System.lineSeparator()
+                + "result," + result;
```

Comment:
let's maybe use StringBuilder?

Reply:
Initially I did but Idea advised me to change to this. Apparently, as per specifications, compiler will use StringBuilder with simple concatenations like this one (not in a loop etc). Also found this [thread](https://intellij-support.jetbrains.com/hc/en-us/community/posts/115000093250-Convert-StringBuilder-to-String-concatenation-suggestion) on the Intellij support forum and it basically confirms that it's equally or more efficient and easier to read. If you still think I should change it I will.

Reply:
Yep, If it's not inside the loop you don't need StringBuilder